tests/test_models.py

```diff
@@ -1,40 +1,171 @@
+
+    def test_update_a_customer(self):
+        """It should Update a Customer"""
+        customer = CustomerFactory()
+        logging.debug(customer)
+        customer.id = None
+        customer.create()
+        logging.debug(customer)
+        self.assertIsNotNone(customer.id)
+        # Change it and save it
+        customer.lastname = "Moreno"
+        original_id = customer.id
+        customer.update()
+        self.assertEqual(customer.id, original_id)
+        self.assertEqual(customer.lastname, "Moreno")
+        # Fetch it back and make sure the id hasn't changed
+        # but the data did change
+        customers = Customer.all()
+        self.assertEqual(len(customers), 1)
+        self.assertEqual(customers[0].id, original_id)
+        self.assertEqual(customers[0].lastname, "Moreno")
+
+    def test_update_no_id(self):
+        """It should not Update a Customer with no id"""
+        customer = CustomerFactory()
+        logging.debug(customer)
+        customer.id = None
+        self.assertRaises(DataValidationError, customer.update)
+    
+    def test_delete_a_customer(self):
+        """It should Delete a Customer"""
+        customer = CustomerFactory()
+        customer.create()
+        self.assertEqual(len(Customer.all()), 1)
+        # delete the customer and make sure it isn't in the database
+        customer.delete()
+        self.assertEqual(len(Customer.all()), 0)
+    
+    def test_list_all_customers(self):
+        """It should List all Customers in the database"""
+        customers = Customer.all()
+        self.assertEqual(customers, [])
+        num_new_customers = 5
+        # Create 5 Customers
+        for _ in range(num_new_customers):
+            customer = CustomerFactory()
+            customer.create()
+        # See if we get back 5 customers
+        customers = Customer.all()
+        self.assertEqual(len(customers), num_new_customers)
+
+    def test_serialize_a_customer(self):
+        """It should serialize a Customer"""
+        customer = CustomerFactory()
+        data = customer.serialize()
+        self.assertEqual(data["id"], customer.id)
+        self.assertEqual(data["firstname"], customer.firstname)
+        self.assertEqual(data["lastname"], customer.lastname)
+        self.assertEqual(data["email"], customer.email)
+        self.assertEqual(data["phone"], customer.phone)
+        self.assertEqual(data["street_line1"], customer.street_line1)
+        self.assertEqual(data["street_line2"], customer.street_line2)
+        self.assertEqual(data["city"], customer.city)
+        self.assertEqual(data["state"], customer.state)
+        self.assertEqual(data["country"], customer.country) 
+        self.assertEqual(data["zipcode"], customer.zipcode) 
+        self.assertEqual(data["created_at"], customer.created_at)  
+        self.assertEqual(data["updated_at"], customer.updated_at)  
+    
+    def test_deserialize_a_customer(self):
```

Comment:
I think you'll need to extend this test to include the other customer values to ensure we don't miss other typos like mentioned above